File: src/main/java/com/manning/javapersistence/ch08/java/model/Bid.java
package com.manning.javapersistence.ch08.java.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Entity
public class Bid {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private BigDecimal amount;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "ITEM_ID", nullable = false)
    private Item item;

    protected Bid() {
    }

    public Bid(BigDecimal amount, Item item) {
        this.amount = amount;
        this.item = item;
    }

    public Long getId() {
        return id;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public Item getItem() {
        return item;
    }

    @Override
    public String toString() {
        return "Bid{id=" + id + ", amount=" + amount + ", itemId=" +
                (item == null ? null : item.getId()) + "}";
    }
}

File: src/main/kotlin/com/manning/javapersistence/ch08/kotlin/model/Bid.kt
package com.manning.javapersistence.ch08.kotlin.model

import java.math.BigDecimal
import javax.persistence.*
import javax.validation.constraints.NotNull

@Entity
class Bid(
    @field:NotNull var amount: BigDecimal? = null,
    @ManyToOne(fetch = FetchType.LAZY) @JoinColumn(name = "ITEM_ID", nullable = false)
    var item: Item? = null
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    override fun toString() = "Bid{id=$id, amount=$amount, itemId=${item?.id}}"
}
